pyqtgraph/opengl/items/GLTextItem.py

```diff
@@ -0,0 +1,90 @@
+from typing import List, Tuple, Union
+from OpenGL.GL import *
+from OpenGL.GLUT import *
+import numpy as np
+from pyqtgraph.opengl.GLGraphicsItem import GLGraphicsItem
+
+class GLTextItem(GLGraphicsItem):
+    """Draws text in 3D."""
+
+    def __init__(self, **kwds):
+        """All keyword arguments are passed to setData()"""
+        GLGraphicsItem.__init__(self)
+        glopts = kwds.pop('glOptions', 'additive')
+        self.setGLOptions(glopts)
+
+        self.pos:Union[np.ndarray, List[float], Tuple[float, ...]] = np.array([0.0, 0.0, 0.0])
+        self.color:Union[np.ndarray, List[float], Tuple[float, ...]] = np.array([1.0, 1.0, 1.0, 1.0])
```

Comment:
We have some mechanisms already in-place to support robust color generation:

```python

import pyqtgraph.functions as fn

...
    def __init__(self, **kwargs):
        ....
        self.color = (1.0, 1.0, 1.0, 1.0)  # use a tuple
        ....

    def setData(self, **kwds):
        ....
        if arg == "color":
            self.color = glColor(value)
        ....
```

The reason we want to use `glColor` here is that we have a _very_ wide-ranging supported color input validation in `pyqtgraph.functions.mkColor` and have a layer that wraps that, `pyqtgraph.functions.glColor` around that.

Reply:
> ```python
> if arg == "color":
>     self.color = glColor(value)
> ```
It's not appropriate because `glColor()` returns None.

In the `__init__` method, `color` is now initialized as a tuple.
The `setData` method has been left unchanged. ([97e39cd](https://github.com/pyqtgraph/pyqtgraph/pull/1776/commits/97e39cde831c710998d9cf8b02064144a1986cf9))

Reply:
I fixed in [2f26a72](https://github.com/pyqtgraph/pyqtgraph/pull/1776/commits/2f26a720647e082cb4638cd267ab55f639165b70).